src/m365/purview/commands/auditlog/auditlog-list.ts

```diff
@@ -0,0 +1,244 @@
+import Auth from '../../../../Auth';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import request, { CliRequestOptions } from '../../../../request';
+import { accessToken } from '../../../../utils/accessToken';
+import { validation } from '../../../../utils/validation';
+import O365MgmtCommand from '../../../base/O365MgmtCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  contentType: string;
+  startTime?: string;
+  endTime?: string;
```

Comment:
Would it make sense to add a top flag that allows you to return say the last 300 items?

Reply:
In which use case can this be useful?
Nevertheless it will be quite hard I guess. While retrieving the audit logs, I have to query a ton of URLs. I cannot pass a `$top` parameter like we are used to in SP REST and Graph.
The annoying thing is also that there logs are not guaranteed to be sorted on creationDate. That's why after retrieving all logs, I order them on creationDate myself.

So adding a top flag will be quite hard to implement I guess.

Reply:
If I have a large tenant with many signals I would like to see what is coming back first ensuring the call doesnt take ages.

If the only way to get sorted values back is through client side, this usecase becomes obsolete. So I have to play around with dates only

Lets keep it as it is

Reply:
The dates are somewhat near eachother but noticed that it's no guarantee that they are sorted by date.
Also all logs are collected via parallel requests (30 GET requests at a time). This also messes up the sorting in creation date. That's why I sort them again on the client side.

Reply:
The dates are somewhat near eachother but noticed that it's no guarantee that they are sorted by date.
Also all logs are collected via parallel requests (30 GET requests at a time). This also messes up the sorting in creation date. That's why I sort them again on the client side.